File: leetcode2025/2025_02/leetcode_1742_countBalls.py
from collections import defaultdict

class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        cnt = defaultdict(int)
        for i in range(lowLimit, highLimit+1):
            tmp = sum(map(int, str(i)))
            cnt[tmp] += 1
        return max(cnt.values())

if __name__ == "__main__":
    s = Solution()
    lowLimit, highLimit = 1, 10
    print(s.countBalls(lowLimit, highLimit))

File: 04.Simulation模拟/leetcode_1742_countBalls.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int countBalls(int lowLimit, int highLimit) {
        unordered_map<int, int> mp;
        for (int i=lowLimit; i<=highLimit; i++){
            int tmp = i, sm = 0;
            while (tmp){
                sm += tmp%10;
                tmp /= 10;
            }
            mp[sm]++;
        }
        int res = 0;
        for (auto& kv : mp) res = max(res, kv.second);
        return res;
    }
};

int main(){
    Solution s;
    int lowLimit=1, highLimit=10;
    cout << s.countBalls(lowLimit, highLimit) << endl;
}
